corehq/apps/hqadmin/reports.py

```diff
@@ -871,25 +881,47 @@ def format_date(dstr, default):
                     dom.get('internal', {}).get('sub_area') or _('No info')
                 ]
 
-    def _calc_num_active_users_per_country(self):
+    def _calc_num_active_users_per_country(self, filters):
         active_users_per_country = (NestedTermAggregationsHelper(
-                                    base_query=DomainES(),
+                                    base_query=DomainES().filter(filters),
                                     terms=[AggregationTerm('countries', 'deployment.countries')],
                                     bottom_level_aggregation=SumAggregation('users', 'cp_n_active_cc_users')
                                     ).get_data())
         return active_users_per_country
 
-    def _calc_num_projs_per_countries(self):
+    def _calc_num_projs_per_countries(self, filters):
         num_projects_by_country = (DomainES()
+                                   .filter(filters)
                                    .terms_aggregation('deployment.countries', 'countries')
                                    .size(0).run().aggregations.countries.counts_by_bucket())
         return num_projects_by_country
 
+    def parse_params(self, es_params):
+        es_filters = {}
+
+        def get_country_params():
```

Comment:
nit but i think it would be clearer (and also have fewer redundant dict lookups) if these were just variable assignments instead of functions.